caddytls/storagetest/memorystorage.go

```diff
@@ -0,0 +1,158 @@
+package storagetest
+
+import (
+	"github.com/mholt/caddy/caddytls"
+	"net/url"
+	"sync"
+)
+
+// memoryMutex is a mutex used to control access to memoryStoragesByCAURL.
+var memoryMutex sync.Mutex
+
+// memoryStoragesByCAURL is a map keyed by a CA URL string with values of
+// instantiated memory stores. Do not access this directly, it is used by
+// InMemoryStorageCreator.
+var memoryStoragesByCAURL = make(map[string]*InMemoryStorage)
+
+// InMemoryStorageCreator is a caddytls.Storage.StorageCreator to create
+// InMemoryStorage instances for testing.
+func InMemoryStorageCreator(caURL *url.URL) (caddytls.Storage, error) {
```

Comment:
I figured this was used for testing like those items in `iotest` and `httptest` are only used for testing. This can be used by other packages for their testing (same with the test harness), ref [this SO answer](http://stackoverflow.com/questions/31794141/can-i-create-shared-test-utilities-in-go)